sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -416,6 +415,97 @@ private object CosmosChangeFeedConfig {
+                                                 targetThroughput: Option[Int],
+                                                 targetThroughputThreshold: Option[Double],
+                                                 globalControlDatabase: String,
+                                                 globalControlContainer: String,
+                                                 globalControlRenewInterval: Option[Duration],
+                                                 globalControlExpireInterval: Option[Duration])
+
+private object CosmosThroughputControlConfig {
+    private val enableThroughputControlSupplier = CosmosConfigEntry[Boolean](
+        key = "spark.cosmos.enableThroughputControl",
+        mandatory = false,
+        defaultValue = Some(false),
+        parseFromStringFunction = enableThroughputControl => enableThroughputControl.toBoolean,
+        helpMessage = "A flag to indicate whether throughput control is enabled.")
+
+    private val groupNameSupplier = CosmosConfigEntry[String](
+        key = "spark.cosmos.throughputControl.name",
+        mandatory = false,
+        parseFromStringFunction = groupName => groupName,
+        helpMessage = "Throughput control group name. " +
+            "Since customer is allowed to create many groups for a container, the name should be unique.")
+
+    private val targetThroughputSupplier = CosmosConfigEntry[Int](
+        key = "spark.cosmos.throughputControl.targetThroughput",
+        mandatory = false,
+        parseFromStringFunction = targetThroughput => targetThroughput.toInt,
+        helpMessage = "Throughput control group target throughput. The value should be larger than 0.")
+
+    private val targetThroughputThresholdSupplier = CosmosConfigEntry[Double](
+        key = "spark.cosmos.throughputControl.targetThroughputThreshold",
+        mandatory = false,
+        parseFromStringFunction = targetThroughput => targetThroughput.toDouble,
+        helpMessage = "Throughput control group target throughput threshold. The value should be between (0,1]. ")
+
+    private val globalControlDatabaseSupplier = CosmosConfigEntry[String](
+        key = "spark.cosmos.throughputControl.globalControl.database",
+        mandatory = false,
+        parseFromStringFunction = globalControlDatabase => globalControlDatabase,
+        helpMessage = "Database which will be used for throughput global control.")
+
+    private val globalControlContainerSupplier = CosmosConfigEntry[String](
+        key = "spark.cosmos.throughputControl.globalControl.container",
+        mandatory = false,
+        parseFromStringFunction = globalControlContainer => globalControlContainer,
+        helpMessage = "Container which will be used for throughput global control.")
+
+    private val globalControlItemRenewIntervalSupplier = CosmosConfigEntry[Duration](
+        key = "spark.cosmos.throughputControl.globalControl.renewIntervalInSeconds",
```

Comment:
regarding the time unit. I am assuming we don't want to expose millisecond granularity as it will create huge overhead.

However, the client connection configs (to be added), will have millisecond granularity.

So I wonder what is the right unit to choose? I see multiple options
1. all configs to be of milli-second unit, for the configs which require less granularity we just extract the second part from the provided milli-second period
2. all configs to be of Duration format ("PT3H4M5S") we throw away the millisecond portion when less granularity is needed (similar to 1)
3. different configs to have different unit.

I personally prefer 1 or 2.

Reply:
changed to option 1.
spark.cosmos.throughputControl.globalControl.renewIntervalInMilliseconds
spark.cosmos.throughputControl.globalControl.expireIntervalInMilliseconds

Reply:
can we change `InMilliseconds` -> `InMS`

Reply:
updated.